### PR TITLE
Move federated sharing routes to the federatedfilesharing app

### PR DESCRIPTION
**Pull request by rullzer**
- Move routes to app
- Move routes over to the AppFramework
- Fix tests

There is plenty of stuff to fix properly here. But out of scope for now.

Lets see what our CI has to say about this.

### CODE REVIEW

**Comment by rullzer** [comment]
Yes! All still seems to work.

Review time: @schiessle @nickvergessen @MorrisJobke @LukasReschke

**Comment by icewind1991** [comment]
Looks good :+1:

File: apps/federatedfilesharing/lib/Controller/RequestHandlerController.php [hunk 1]
```diff
@@ -71,41 +71,48 @@ class RequestHandler {
...
 		$this->connection = $connection;
 		$this->shareManager = $shareManager;
-		$this->request = $request;
 		$this->notifications = $notifications;
 		$this->addressHandler = $addressHandler;
 		$this->userManager = $userManager;
 	}
 
 	/**
+	 * @NoAdminRequired
```
Comment (LukasReschke) [inline apps/federatedfilesharing/lib/Controller/RequestHandlerController.php]: PublicPage implies that

Comment (rullzer) [inline apps/federatedfilesharing/lib/Controller/RequestHandlerController.php]: Fixed

**Comment by MorrisJobke** [comment]
Tested and works (with existing federated storages and new ones) 👍

**Pull request merged**